src/json.cpp

```diff
@@ -1474,10 +1474,11 @@ void JsonIn::error(std::string message, int offset)
     size_t pos = tell();
     rewind(3, 240);
     size_t startpos = tell();
-    char buffer[241];
-    stream->read(&buffer[0], pos - startpos);
+    char *buffer = new char[pos - startpos + 1];
```

Comment:
Would be even better without raw memory allocation, `buffer` could be a string for example. Initialized as `std::string buffer( pos - startpos, '\0' );` - note that this does not need the terminating 0-char (and no `+ 1` to store it) as the string knows its length. It also avoids the need to `delete` the buffer.

Btw. `delete buffer;` invokes undefined behavior, the pointer was created using `new []`, and must therefor be freed using `delete []`. But using a string will neatly avoid this problem.